src/chocolatey.benchmark/helpers/PinvokeProcessHelper.cs

```diff
@@ -0,0 +1,472 @@
+                tree.Processes.AddLast(nextProcess.ProcessName);
+            }
+
+            return tree;
+        }
+
+        public static string GetUndocumentedParent(Process process = null)
+        {
+            if (Platform.GetPlatform() != PlatformType.Windows)
+            {
+                return null;
+            }
+
+            if (process == null)
+            {
+                process = Process.GetCurrentProcess();
+            }
+
+            Process nextProcess = null;
+
+            while (true)
+            {
+                var foundProcess = ParentProcessUtilities.GetParentProcess(nextProcess ?? process);
+
+                if (foundProcess == null)
+                {
+                    break;
+                }
+
+                nextProcess = foundProcess;
+            }
+
+            return nextProcess?.ProcessName;
+
+            //var parentProcess = ParentProcessUtilities.GetParentProcess(process);
+
+            //if (parentProcess == null)
+            //{
+            //    return null;
+            //}
+
+            //var topLevelProcess = GetUndocumentedParent(parentProcess);
+
+            //if (topLevelProcess == null)
+            //{
+            //    return parentProcess.ProcessName;
+            //}
+
+            //return topLevelProcess;
+        }
+
+        public static string GetUndocumentedParentFiltered(Process process = null)
+        {
+            if (Platform.GetPlatform() != PlatformType.Windows)
+            {
+                return null;
+            }
+
+            if (process == null)
+            {
+                process = Process.GetCurrentProcess();
+            }
+
+            Process nextProcess = null;
+            Process selectedProcess = null;
+
+            while (true)
+            {
+                var foundProcess = ParentProcessUtilities.GetParentProcess(nextProcess ?? process);
+
+                if (foundProcess == null)
+                {
+                    break;
+                }
+
+                nextProcess = foundProcess;
+
+                if (!IsIgnoredParent(nextProcess.ProcessName))
+                {
+                    selectedProcess = nextProcess;
+                }
+            }
+
+            return selectedProcess?.ProcessName;
+
+            //var parentProcess = ParentProcessUtilities.GetParentProcess(process);
+
+            //if (parentProcess == null)
+            //{
+            //    return null;
+            //}
+
+            //var topLevelProcess = GetUndocumentedParentFiltered(parentProcess);
+
+            //if (topLevelProcess == null || IsIgnoredParent(topLevelProcess))
+            //{
+            //    if (IsIgnoredParent(parentProcess.ProcessName))
+            //    {
+            //        return null;
+            //    }
+            //    else
+            //    {
+            //        return parentProcess.ProcessName;
+            //    }
+            //}
+
+            //return topLevelProcess;

```

Comment:
Are these comments needed?

Reply:
Same as above, it can be safely removed.